int16 rweight
int16 mrweight
int16 mlweight
int16 lweight
